File: docs/snippets/Datamaps/datamap/create-datamap.py
# docs:begin
from arctrl import DataContext, Datamap

leaf_area = DataContext(
    None,
    "assays/Measurement/dataset/results.csv",
    None,
    None,
    None,
    None,
    None,
    None,
    "leaf area",
    "Leaf area measurements exported from image analysis.",
)

datamap = Datamap([leaf_area])

first_context = datamap.GetDataContext(0)
# docs:end

# docs:assert
if len(datamap.DataContexts) != 1:
    raise Exception(f"Expected one data context, got {len(datamap.DataContexts)}")

if first_context.Name != "assays/Measurement/dataset/results.csv":
    raise Exception("Expected the data context path to be stored as its name.")
# docs:endassert

File: docs/snippets/Datamaps/datamap/create-datamap.ts
// docs:begin
import { DataContext, Datamap } from "@nfdi4plants/arctrl";

const leafArea = new DataContext(
  undefined,
  "assays/Measurement/dataset/results.csv",
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  "leaf area",
  "Leaf area measurements exported from image analysis.",
);

const datamap = new Datamap([leafArea]);

const firstContext = datamap.GetDataContext(0);
// docs:end

// docs:assert
if (datamap.DataContexts.length !== 1) {
  throw new Error(`Expected one data context, got ${datamap.DataContexts.length}`);
}

if (firstContext === undefined) {
  throw new Error("Expected the first data context to be readable.");
}
// docs:endassert
